### PR TITLE
Add spellcheck config

### PR DESCRIPTION
**Pull request by lukashass**
Part of #738 

```
pnpx cspell lint --gitignore '{**,.*}/{*,.*}'
```

### CODE REVIEW

**Comment by anbraten** [comment]
CI job works now: https://ci.woodpecker-ci.org/repos/3780/pipeline/11011/33
Just "some" typos left.

**Comment by pat-s** [comment]
@lukashass Do you want to clean up here? :)

**Comment by lukashass** [comment]
Yes, its just a lot :see_no_evil:

File: .cspell.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "xormigrate",
+    "zerolog",
+    "Msgf"
+  ],
+  "ignorePaths": [
+    "**/node_modules/**/*",
+    "pnpm-lock.yaml",
+    ".cspell.json",
+    "web/src/assets/locales/**/*",
+    "server/store/datastore/migration/testfiles/sqlite.db",
```
Comment (6543) [inline .cspell.json:28]:
```suggestion
    "server/store/datastore/migration/**/*",
```

Comment (anbraten) [reply .cspell.json]: Why should the code / other files in migrations not be checked? I am pretty sure this was intentional. CC @lukashass

**Comment by anbraten** [comment]
> Yes, its just a lot 🙈

We could split it by ignoring paths first and for example just correct the `agent/` folder in this PR. This way others could correct code as well and we could simply open new PRs removing the excluded area with all translation fixes.

**Comment by lukashass** [comment]
> We could split it by ignoring paths first and for example just correct the `agent/` folder in this PR. This way others could correct code as well and we could simply open new PRs removing the excluded area with all corresponding translation fixes.

Most of the work to be done is adding words to the allow list, so this wouldn't work very well. I'll have some time next week to continue with this.

**Comment by lukashass** [comment]
I decided to do this in steps after all. So `web/` is done, others are started, but ignored for now.

File: .cspell.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    ".cspell.json",
+    ".golangci.yaml",
+    ".vscode/extensions.json",
+    "web/src/assets/locales/**/*",
+    "web/components.d.ts",
+    "server/store/datastore/migration/**/*",
+    "go.mod",
+    "go.sum",
+    "*.excalidraw",
+    "CHANGELOG.md",
```
Comment (anbraten) [inline .cspell.json:98]: Should probably be also be checked in the future. It will fail in the release PR so we will know if we need to rename a PR.

**Review by anbraten — APPROVED** [review]

File: web/src/components/atomic/Icon.vue [hunk 3]
```diff
@@ -7,7 +8,7 @@
   <i-mdi-gesture-tap v-else-if="name === 'manual-pipeline'" class="h-6 w-6" />
   <i-mdi-tag-outline v-else-if="name === 'tag'" class="h-6 w-6" />
   <i-clarity-deploy-line v-else-if="name === 'deployment'" class="h-6 w-6" />
-  <i-mdisource-commit v-else-if="name === 'commit'" class="h-6 w-6" />
+  <i-mdi-source-commit v-else-if="name === 'commit'" class="h-6 w-6" />
```
Comment (6543) [inline web/src/components/atomic/Icon.vue:11]: was this a bug? does that icon exist?

File: .gitpod.yml [hunk 4]
```diff
@@ -105,6 +105,7 @@ ports:
 
 vscode:
   extensions:
+    # cSpell:disable
```
Comment (6543) [inline .gitpod.yml:108]: ?

File: .cspell.json (hunk 2 above)
Comment (lukashass) [reply .cspell.json:98]: Checking commit messages should probably be done before merging PRs then. Will move it down to the TODOs

File: web/src/components/atomic/Icon.vue (hunk 3 above)
Comment (lukashass) [reply web/src/components/atomic/Icon.vue:11]: I think both work, with the `-` even the inline preview of my VSCode plugin works, but not tested in the app.

File: .gitpod.yml (hunk 4 above)
Comment (lukashass) [reply .gitpod.yml:108]: Extensions have mostly usernames, not really worth spell checking IMO

Comment (6543) [reply .gitpod.yml:108]: No you have two comments here, should ir not be one?

Comment (lukashass) [reply .gitpod.yml:108]: Ah the second comment enables cSpell for any lines that might be added below

File: .cspell.json [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  ],
+  "ignorePaths": [
+    "**/node_modules/**/*",
+    "pnpm-lock.yaml",
+    ".gitignore",
+    ".git/**/*",
+    ".cspell.json",
+    ".golangci.yaml",
+    ".vscode/extensions.json",
+    "web/src/assets/locales/**/*",
```
[the remark this answers is not in the document]
Comment (lukashass) [reply .cspell.json:92]: I'm not sure if this is possible with the glob library cSpell is using: https://cspell.org/docs/globs/

> Glob patterns used to exclude files and directories from being checked. ! are only partially supported due to an issue with the Glob library[1](https://github.com/isaacs/node-glob/issues/409)

File: .gitpod.yml (hunk 4 above)
Comment (6543) [reply .gitpod.yml:108]: Diff looked straange

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
Thanks :)

**Pull request merged**